test/test_transforms_v2_refactored.py

```diff
@@ -1833,6 +1838,96 @@ def call_transform():
             assert output[0] is image
             assert output[1] is label
 
+    def test_compose(self):
+        transform = transforms.Compose(
+            [
+                transforms.RandomHorizontalFlip(p=1),
+                transforms.RandomVerticalFlip(p=1),
+            ]
+        )
+
+        input = make_image()
+
+        actual = check_transform(transform, input)
+        expected = F.vertical_flip(F.horizontal_flip(input))
+
+        assert_equal(actual, expected)
+
+    @pytest.mark.parametrize("p", [0.0, 1.0])
+    @pytest.mark.parametrize("sequence_type", [list, nn.ModuleList])
+    def test_random_apply(self, p, sequence_type):
+        transform = transforms.RandomApply(
+            sequence_type(
+                [
+                    transforms.RandomHorizontalFlip(p=1),
+                    transforms.RandomVerticalFlip(p=1),
+                ]
+            ),
+            p=p,
+        )
+
+        # This needs to be a pure tensor (or a PIL image), because otherwise check_transforms skips the v1 compatibility
+        # check
+        input = make_image_tensor()
+        output = check_transform(transform, input, check_v1_compatibility=issubclass(sequence_type, nn.ModuleList))
+
+        if p == 1:
+            assert_equal(output, F.vertical_flip(F.horizontal_flip(input)))
+        else:
+            assert output is input
+
+    @pytest.mark.parametrize("p", [(0, 1), (1, 0)])
+    def test_random_choice(self, p):
+        transform = transforms.RandomChoice(
+            [
+                transforms.RandomHorizontalFlip(p=1),
+                transforms.RandomVerticalFlip(p=1),
+            ],
+            p=p,
+        )
+
+        input = make_image()
+        output = check_transform(transform, input)
+
+        p_horz, p_vert = p
+        if p_horz:
+            assert_equal(output, F.horizontal_flip(input))
+        else:
+            assert_equal(output, F.vertical_flip(input))
+
+    def test_random_order(self):
+        transform = transforms.Compose(
+            [
+                transforms.RandomHorizontalFlip(p=1),
+                transforms.RandomVerticalFlip(p=1),
+            ]
+        )
+
+        input = make_image()
+
+        actual = check_transform(transform, input)
+        # horizontal and vertical flip are commutative. Meaning, although the order in the transform is indeed random,
+        # we don't need to care here.
+        expected = F.vertical_flip(F.horizontal_flip(input))
+
+        assert_equal(actual, expected)
```

Comment:
Unfortunately it also means that this test doesn't check that the order of the transforms is indeed random. At best it checks that the input transforms are applied. I think we should at least acknowledge that in the comment instead of saying "we don't need to care".

Reply:
Fair enough. All ears if you have an idea to check whether the transforms are actually applied in random order.

Reply:
a pair of non-commutative transforms, with 2 hand-chosen seeds, asserting that the results are different. What makes it a bit harder is that the transforms themselves must be non-random. Worst case scenario we could just define `Plus2` and `Times2`.